python/tests/test_tables.py

```diff
@@ -2540,6 +2539,88 @@ def test_sequence_length_longer_than_edges(self):
         assert len(tree.parent_dict) == 0
 
 
+class TestEqualityOptions:
+    def test_equals_provenance(self):
+        t1 = msprime.simulate(10, random_seed=42).tables
+        t2 = msprime.simulate(10, random_seed=42).tables
```

Comment:
Windows is having trouble here as the resolution of `datetime.datetime.now()` is about 1ms(!!)
```suggestion
        t1 = msprime.simulate(10, random_seed=42).tables
        time.sleep(0.1)
        t2 = msprime.simulate(10, random_seed=42).tables
```